src/wwwroot/app/services/VacancyService.js

```diff
@@ -68,22 +67,29 @@ export default class VacancyService {
       vacancy = _convertFromServerDates(vacancy);
       let userPromise = _fillUser(vacancy);
       vacancy.responsibleId = toString(vacancy.responsibleId);
+      vacancy = _VacancyService._fillChildVacancies(vacancy);
       return _$q.all([_fillThesauruses(vacancy), userPromise]).then(first);
    }
 
    remove(vacancy) {
-      if (vacancy.id) {
-         const additionalUrl = VACANCY_URL + vacancy.id;
-         return _HttpService.remove(additionalUrl, vacancy);
+      if (vacancy.id && vacancy.parentVacancyId !== null) {
+         let predicate = {id: vacancy.id};
+         let parentVacancy = {};
+         return _VacancyService.getVacancy(vacancy.parentVacancyId).then((responseVacancy) => {
+            parentVacancy = responseVacancy;
+            return _HttpService.remove(`${VACANCY_URL}${vacancy.id}`, vacancy).then(() => {
```

Comment:
unwrap this sequence in the next way: you return the promise result to scope of getVacancy call, here you use your then of removement